dot/digest/digest.go

```diff
@@ -111,9 +109,8 @@ func (h *Handler) Start() error {
 // Stop stops the Handler
 func (h *Handler) Stop() error {
 	h.cancel()
-	h.blockState.UnregisterImportedChannel(h.importedID)
+	h.blockState.FreeImportedBlockNotifierChannel(h.imported)
 	h.blockState.UnregisterFinalisedChannel(h.finalisedID)
-	close(h.imported)
 	close(h.finalised)
```

Comment:
it would be good to update the `finalised` notifiers with the new architecture as well, as it has the same problem. can open another issue or do this in this PR, up to you

Reply:
Yes, I'll update once the architecture is set.